### PR TITLE
added my fork of grpc-caller

### PR DESCRIPTION
**Pull request by electricmonk**

### CODE REVIEW

File: projects/common-library-typescript/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "dependencies": {
     "bs58": "^4.0.1",
     "grpc": "^1.8.4",
-    "grpc-caller": "^0.4.0",
+    "grpc-caller": "https://github.com/electricmonk/grpc-caller",
```
Comment (idandash) [inline projects/common-library-typescript/package.json]: Much safer to enforce a specific version (aka git version tag)

Comment (electricmonk) [reply projects/common-library-typescript/package.json]: why? it's my fork

**Comment by electricmonk** [comment]
@lbeder not sure what you mean by using `Metadata`; what I aim to achieve here is to have the client add extra data to the error and to alter its message. `Metadata` seems to be a way to transfer context from client to server, i.e. authentication or other stuff that you would want to propagate between servers

File: projects/common-library-typescript/package.json (hunk 1 above)
Comment (idandash) [reply projects/common-library-typescript/package.json]: Sure. but even your fork might be updated at some point.

Comment (electricmonk) [reply projects/common-library-typescript/package.json]: whereupon I'll want the project to get the most recent version, no?

**Comment by lbeder** [comment]
@electricmonk it feels too excessive to me. Why can't we add this logic externally, by wrapping our `grpc` abstraction layer with `try/catch`? Forking such a major package, just for that, thus forcing us to start syncing it explicitly with the mainstream. feels like an overkill.

**Comment by electricmonk** [comment]
Dude. It’s not grpc. It’s a small wrapper.

Shai
________________________________

**Pull request merged**